simd/src/Kokkos_SIMD_AVX2.hpp

```diff
@@ -724,25 +724,48 @@ class basic_simd<double, simd_abi::avx2_fixed_size<4>> {
   }
   template <typename FlagType>
   KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION constexpr explicit basic_simd(
-      value_type const* ptr, FlagType flag) {
-    copy_from(ptr, flag);
+      const value_type* ptr, FlagType) noexcept {
+    if constexpr (std::is_same_v<FlagType,
+                                 simd_flags<simd_alignment_vector_aligned>>) {
+      m_value = _mm256_load_pd(ptr);
+    } else {
+      m_value = _mm256_loadu_pd(ptr);
+    }
   }
+  template <typename FlagType>
+  KOKKOS_IMPL_HOST_FORCEINLINE_FUNCTION constexpr explicit basic_simd(
+      const value_type* ptr, mask_type const& mask, FlagType) noexcept {
+#ifdef KOKKOS_IMPL_WORKAROUND_ROCM_AVX2_ISSUE
+    __m256d tmp = _mm256_load_pd(ptr);
+    m_value     = _mm256_and_pd(tmp, static_cast<__m256d>(mask));
+#else
+    m_value = _mm256_maskload_pd(
+        ptr, _mm256_castpd_si256(static_cast<__m256d>(mask)));
+#endif
```

Comment:
`KOKKOS_IMPL_WORKAROUND_ROCM_AVX2_ISSUE` has been removed in https://github.com/kokkos/kokkos/pull/8401
```suggestion
    m_value = _mm256_maskload_pd(
        ptr, _mm256_castpd_si256(static_cast<__m256d>(mask)));
```